### PR TITLE
Add Koolertron AMAG09 9 key macropad

### PR DESCRIPTION
**Pull request by al177**
Added default and vial keymaps for the Koolertron AMAG09. This is a cheap and available board based on an ATMEGA32A that ships with a rebranded Vial firmware and GUI (MYKB).

### CODE REVIEW

File: keyboards/koolertron/amag09/amag09.h [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include "quantum.h"
+
+// clang-format off
+#define LAYOUT_all( \
```
Comment (lesshonor) [inline keyboards/koolertron/amag09/amag09.h]: If you've only got one layout, you can just use `LAYOUT`.

```suggestion
#define LAYOUT( \
```
By the way, you probably want to move this layout/matrix info into `info.json` and then delete this file. This is one of those things that works now, but will eventually break. [You can see other examples of how this works in the repository](https://github.com/vial-kb/vial-qmk/blob/47721416a6483a1d759c21a18beae3c0cb7d3eb2/keyboards/jacky_studio/piggy60/info.json), and/or run `qmk info -f json -kb koolertron/amag09` for an idea of the syntax (though that command ends up with some duplicate data)

File: keyboards/koolertron/amag09/config.h [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include "config_common.h"
+
```
Comment (lesshonor) [inline keyboards/koolertron/amag09/config.h]: Don't need to include `config_common.h`.
```suggestion
```
You can move the matrix pins, diode direction and backlight pin into `info.json` too.

File: keyboards/koolertron/amag09/config.h [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+#define MATRIX_COL_PINS { B0, B1, B2 }
+
+/* COL2ROW, ROW2COL*/
+#define DIODE_DIRECTION ROW2COL
+
+#define BACKLIGHT_PIN D4
+
+/* Debounce reduces chatter (unintended double-presses) - set 0 if debouncing is not needed */
+#define DEBOUNCE 5
+
```
Comment (lesshonor) [inline keyboards/koolertron/amag09/config.h]: This is the default, so you don't need to specify it.
```suggestion
```

File: keyboards/koolertron/amag09/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+enum layer_names {
+    BASE,
+    FN1,
+    FN2,
+    FN3
+};
+
+// clang-format off
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [BASE] = LAYOUT_all(
```
Comment (lesshonor) [inline keyboards/koolertron/amag09/keymaps/default/keymap.c]:
```suggestion
  [BASE] = LAYOUT(
```

File: keyboards/koolertron/amag09/keymaps/default/keymap.c [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+      KC_KP_7, KC_KP_8, KC_KP_9,
+      KC_KP_4, KC_KP_5, KC_KP_6,
+      KC_KP_1, KC_KP_2, KC_KP_3
+  ),
+  [FN1] = LAYOUT_all(
```
Comment (lesshonor) [inline keyboards/koolertron/amag09/keymaps/default/keymap.c]:
```suggestion
  [FN1] = LAYOUT(
```

File: keyboards/koolertron/amag09/keymaps/default/keymap.c [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+      _______, _______, _______,
+      _______, _______, _______,
+      _______, _______, _______
+  ),
+  [FN2] = LAYOUT_all(
```
Comment (lesshonor) [inline keyboards/koolertron/amag09/keymaps/default/keymap.c]:
```suggestion
  [FN2] = LAYOUT(
```

File: keyboards/koolertron/amag09/keymaps/default/keymap.c [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+      _______, _______, _______,
+      _______, _______, _______,
+      _______, _______, _______
+  ),
+  [FN3] = LAYOUT_all(
```
Comment (lesshonor) [inline keyboards/koolertron/amag09/keymaps/default/keymap.c]:
```suggestion
  [FN3] = LAYOUT(
```

File: keyboards/koolertron/amag09/keymaps/vial/keymap.c [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+enum layer_names {
+    BASE,
+    FN1,
+    FN2,
+    FN3
+};
+
+// clang-format off
+const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
+  [BASE] = LAYOUT_all(
```
Comment (lesshonor) [inline keyboards/koolertron/amag09/keymaps/vial/keymap.c]:
```suggestion
  [BASE] = LAYOUT(
```

File: keyboards/koolertron/amag09/keymaps/vial/keymap.c [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+      KC_KP_7, KC_KP_8, KC_KP_9,
+      KC_KP_4, KC_KP_5, KC_KP_6,
+      MO(FN1), KC_KP_2, KC_KP_3
+  ),
+  [FN1] = LAYOUT_all(
```
Comment (lesshonor) [inline keyboards/koolertron/amag09/keymaps/vial/keymap.c]:
```suggestion
  [FN1] = LAYOUT(
```

File: keyboards/koolertron/amag09/keymaps/vial/keymap.c [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+      BL_TOGG, _______, _______,
+      _______, _______, _______,
+      _______, _______, _______
+  ),
+  [FN2] = LAYOUT_all(
```
Comment (lesshonor) [inline keyboards/koolertron/amag09/keymaps/vial/keymap.c]:
```suggestion
  [FN2] = LAYOUT(
```

File: keyboards/koolertron/amag09/keymaps/vial/keymap.c [hunk 11]
```diff
@@ -0,0 +1,49 @@
...
+      _______, _______, _______,
+      _______, _______, _______,
+      _______, _______, _______
+  ),
+  [FN3] = LAYOUT_all(
```
Comment (lesshonor) [inline keyboards/koolertron/amag09/keymaps/vial/keymap.c]:
```suggestion
  [FN3] = LAYOUT(
```

File: keyboards/koolertron/amag09/amag09.c
Comment (lesshonor) [inline keyboards/koolertron/amag09/amag09.c:1]: You can delete this file.

File: keyboards/koolertron/amag09/rules.mk
Comment (lesshonor) [inline keyboards/koolertron/amag09/rules.mk:1]: You could move all this into `info.json` too. Like most of my other suggestions, this is something that works fine now but will probably break eventually.

File: keyboards/koolertron/amag09/keymaps/vial/config.h [hunk 12]
```diff
@@ -0,0 +1,11 @@
...
+
+#pragma once
+
+#define VIAL_KEYBOARD_UID {0x1C, 0xA6, 0x8B, 0xCD, 0xCB, 0x42, 0x5C, 0x6F}
+
+
+#define VIAL_UNLOCK_COMBO_ROWS { 0, 2 }
+#define VIAL_UNLOCK_COMBO_COLS { 2, 0 }
+
+/* VIAL_INSECURE = yes */
```
Comment (lesshonor) [inline keyboards/koolertron/amag09/keymaps/vial/config.h]:
```suggestion
```

**Review by lesshonor — CHANGES_REQUESTED** [review]

File: keyboards/koolertron/amag09/amag09.h (hunk 1 above)
Comment (al177) [reply keyboards/koolertron/amag09/amag09.h]: Done.

File: keyboards/koolertron/amag09/config.h (hunk 3 above)
Comment (al177) [reply keyboards/koolertron/amag09/config.h]: Done.

File: keyboards/koolertron/amag09/config.h (hunk 2 above)
Comment (al177) [reply keyboards/koolertron/amag09/config.h]: Yep, moved everything from amag09.[ch] and config.h to info.json

File: keyboards/koolertron/amag09/amag09.c
Comment (al177) [reply keyboards/koolertron/amag09/amag09.c:1]: Done

File: keyboards/koolertron/amag09/rules.mk
Comment (al177) [reply keyboards/koolertron/amag09/rules.mk:1]: Thanks for the review and tips!

**Review by lesshonor — APPROVED** [review]

**Pull request merged**